core/src/kmx/kmx_plus.cpp

```diff
@@ -207,22 +226,36 @@ COMP_KMXPLUS_STRS::valid(KMX_DWORD _kmn_unused(length)) const {
     return false;
   }
   for (KMX_DWORD i=0; i<this->count; i++) {
-    const size_t MYBUFSIZ = 256;
-    KMX_WCHAR buf[MYBUFSIZ];
     KMXPLUS_PRINTF(("#0x%X: ", i));
-    PKMX_WCHAR str = this->get(i, buf, MYBUFSIZ);
-    if (!str) {
-        KMXPLUS_PRINTF(("NULL/ERR\n"));
-        return false;
+    KMX_DWORD offset = entries[i].offset;
+    KMX_DWORD length = entries[i].length;
+    if(offset+((length+1)*2) > header.size) {
+      debug_println("expected end of string past header.size");
+      return false;
+    }
+    const uint8_t* thisptr = reinterpret_cast<const uint8_t*>(this);
+    const KMX_WCHAR* start = reinterpret_cast<const KMX_WCHAR*>(thisptr+offset);
+    if(start[length] != 0) {
+      debug_println("String not null terminated");
+      return false;
     }
-    for(int j=0; str[j] && j<0x30; j++) {
-        if (str[j] < 0x7F && str[j] > 0x20) {
-            KMXPLUS_PRINTF(("%c", str[j]));
+    // TODO-LDML: validate valid UTF-16LE?
+#if KMXPLUS_DEBUG
+    // first print it escaped
+    for(KMX_DWORD j=0; start[j] && j<length; j++) {
+        if (start[j] < 0x7F && start[j] > 0x20) {
+            KMXPLUS_PRINTF(("%c", start[j]));
         } else {
-            KMXPLUS_PRINTF((" U+%04X ", str[j]));
+            KMXPLUS_PRINTF((" U+%04X ", start[j]));
         }
     }
     KMXPLUS_PRINTF(("\n"));
+
+    // // now print it as a string
+    // TODO-LDML
+    // std::u16string str = get(i);
+    // std::cerr << str << std::endl;
```

Comment:
todo, figure out how to print u16strings

Reply:
Yep, painful...